packages/teams-js/src/internal/appHelpers.ts

```diff
@@ -70,8 +87,37 @@ export function notifyFailureHelper(apiVersiontag: string, appInitializationFail
   ]);
 }
 
-export function notifySuccessHelper(apiVersionTag: string): void {
-  sendMessageToParent(apiVersionTag, app.Messages.Success, [version]);
+function supportsNotifySuccessResponse(): boolean {
+  return ensureInitialized(runtime) && !!runtime.supports.app?.notifySuccessResponse;
+}
+
+export async function notifySuccessHelper(apiVersionTag: string): Promise<NotifySuccessResponse> {
```

Comment:
It would be good for this comment to have a function explaining what it is doing conceptually, something about how it lets a dev call notifySuccess even while initialize is pending receiving a response, even though the runtime object is required. I want to make sure that people understand the cleverness in here and don't accidentally mess it up in the future

Reply:
I'll add the comment 😄